lib/Conversion/TorchToStablehlo/GatherScatter.cpp

```diff
@@ -507,6 +507,286 @@ LogicalResult ConvertAtenOp<AtenScatterSrcOp>::matchAndRewrite(
+        op, "Only tensor types input are currently supported");
+
+  // Deal with torch.prim.ListConstruct of non const value to get the index
+  auto tensorList = op.getIndices();
+  SmallVector<Value> tensorsTorchType;
+  if (!getListConstructElements(tensorList, tensorsTorchType))
+    return op.emitError(
+        "unimplemented: the tensor list is not from list construct");
+  auto indexTensors = getTypeConvertedValues(
+      rewriter, op->getLoc(), getTypeConverter(), tensorsTorchType);
+
+  // convert list of indices with none into indices tensor  without none
+  // indexTensors (none,[1,2,3]) -> ([0,0,0],[1,2,3])
+  // ([[0],[0],[0]],[[1],[2],[3]])-> [[0,1],[0,2], [0,3]]
+  if (indexTensors.size() <= 1) {
+    return rewriter.notifyMatchFailure(
+        op, "Only support indexput with multiple index.");
+  }
+  SmallVector<Value> indicesConcatTensors;
+  SmallVector<int64_t> indexesRank;
+  SmallVector<SmallVector<int64_t>> indexesShape;
+
+  // concat index tensor into to indices tensor for concat
+  for (size_t i = 0; i < indexTensors.size(); i++) {
+    auto index = indexTensors[i];
+    auto indexTorch = tensorsTorchType[i];
+    // TODO add support for none index other than i==0, like (index0, None)
+    // (None, index1)
+    if (i == 0 && indexTorch.getType().isa<Torch::NoneType>()) {
+      // convert None to [0,0,0]
+      auto indexNext = indexTensors[i + 1];
+      auto indexNextTorch = tensorsTorchType[i + 1];
+      if (indexNextTorch.getType().isa<Torch::NoneType>()) {
+        return rewriter.notifyMatchFailure(
+            op, "Multiple None index is not support for now.");
+      }
+      auto indexNextType = indexNext.getType().dyn_cast<RankedTensorType>();
+      auto indexNextShape = indexNextType.getShape();
+
+      int64_t size = 1;
+      for (auto s : indexNextShape)
+        size *= s;
+      SmallVector<int64_t> values(size, i);
+      index = hlo::getConstTensor<int64_t>(rewriter, op, values, indexNextShape)
+                  .value();
+    }
+
+    auto indexType = index.getType().dyn_cast<RankedTensorType>();
+    // Reshape the zero rank index to rank 1
+    if (indexType.getRank() == 0) {
+      // [] -> [0]
+      SmallVector<int64_t, 1> oneShape({1}); // {1}
+      RankedTensorType reshapeType =
+          RankedTensorType::get(oneShape, indexType.getElementType());
+      auto indexOneReshapeOp =
+          rewriter.create<stablehlo::ReshapeOp>(loc, reshapeType, fillValues);
+      index = indexOneReshapeOp.getResult();
+      indexType = index.getType().dyn_cast<RankedTensorType>();
+    }
+    auto indexShape = indexType.getShape();
+    indexesShape.push_back(makeShapeTorchCompatible(indexShape));
+    indexesRank.push_back(indexType.getRank());
+
+    // Expand last dim of index to indices [3] -> [3,1]
+    // convert [0,0,0]  to [[0],[0],[0]]
+    SmallVector<int64_t> indiceShapeOneDim;
+    for (auto shape : indexShape) {
+      indiceShapeOneDim.push_back(shape);
+    }
```

Comment:
Nit: let's use a constructor like `SmallVector<> a(b.begin(), b.end())` instead of iteratively push_back.